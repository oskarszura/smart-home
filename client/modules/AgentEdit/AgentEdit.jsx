import React from 'react';
import PropTypes from 'prop-types';
import { withRouter } from 'react-router';

class AgentEdit extends React.Component {
  constructor(props) {
    super(props);
    this.updateTemperature = this.updateTemperature.bind(this);
    this.updateConfig = this.updateConfig.bind(this);
  }

  componentDidMount() {
    const {
      agentID,
      fetchConfig,
    } = this.props;

    fetchConfig(agentID);
  }

  updateConfig() {
    const {
      agentID,
      config,
      updateConfig,
    } = this.props;

    updateConfig(agentID, config);
  }

  updateTemperature(e) {
    const temperature = e.target.value;
    const {
      agentID,
      updateTemperature,
    } = this.props;

    updateTemperature(agentID, temperature);
  }

  render() {
    const {
      config: {
        temperature,
      },
    } = this.props;

    return (<div>
      <div className="c-input">
        <div className="c-input__label">
          Temperature modifier
        </div>
        <input
          className="c-input__field"
          value={temperature}
          onChange={this.updateTemperature}
        />
      </div>
      <button
        className="c-btn c-btn--accept"
        onClick={this.updateConfig}
      >
        UPDATE
      </button>
    </div>);
  }
}

AgentEdit.propTypes = {
  config: PropTypes.shape(),
  agentID: PropTypes.string,
  updateTemperature: PropTypes.func,
  fetchConfig: PropTypes.func,
  updateConfig: PropTypes.func,
};

export default withRouter(AgentEdit);
