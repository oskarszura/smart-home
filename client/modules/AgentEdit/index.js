// @flow
import { connect } from 'react-redux';
import * as actions from 'client/models/agentConfigs/actions';
import * as selectors from 'client/models/agentConfigs/selectors';
import * as agentQueries from 'client/models/agents/queries';
import AgentEdit from './AgentEdit';

const mapStateToProps = (state, ownProps) => {
  const {
    match: {
      params,
    },
  } = ownProps;
  const agentID = params.agent;
  const agentConfig = selectors.getAgentConfigById(state, agentID) || {};
  const agent = agentQueries.getAgentById(state, agentID);

  return {
    timestamp: new Date(),
    agent,
    agentConfig
  };
};

const mapDispatchToProps = dispatch => ({
  updateProperty: (agentID, key, value) => {
    dispatch(actions.updateProperty(agentID, key, value));
  },
  commitConfig: (agentID, config) => {
    dispatch(actions.commitAgentConfig(agentID, config));
  },
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(AgentEdit);
