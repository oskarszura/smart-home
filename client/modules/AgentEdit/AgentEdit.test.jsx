// @flow
import _ from 'lodash';
import React from 'react';
import { shallow } from 'enzyme';
import AgentEdit from './AgentEdit';

const agent = {
  id: '90346453',
  name: 'Livingroom',
  data: {},
  type: 'type1-v0.3.3',
  ip: '192.168.1.10',
  isOnline: true,
};

const agentConfig = {
  id: '90346453',
  name: 'Livingroom',
  temperature: 0,
};

describe('AgentEdit', () => {
  it('should render correctly', () => {
    const component = shallow(
      <AgentEdit
        agent={agent}
        agentConfig={agentConfig}
      />
    );

    expect(component).toMatchSnapshot();
  });
});
