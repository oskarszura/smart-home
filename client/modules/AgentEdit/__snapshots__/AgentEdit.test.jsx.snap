// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AgentEdit should render correctly 1`] = `
<Route>
  <Component />
</Route>
`;
